airbyte-config/config-persistence/src/main/java/io/airbyte/config/persistence/ConfigRepository.java

```diff
@@ -615,40 +615,73 @@ public List<StandardSync> listStandardSyncs() throws ConfigNotFoundException, IO
 
   public List<StandardSync> listStandardSyncsUsingOperation(final UUID operationId)
       throws IOException {
-    final Result<Record> result = database.query(ctx -> ctx.select(CONNECTION.asterisk())
+    final Result<Record> connectionResult = database.query(ctx -> ctx.select(CONNECTION.asterisk(), CONNECTION_OPERATION.asterisk())
         .from(CONNECTION)
         .join(CONNECTION_OPERATION)
         .on(CONNECTION_OPERATION.CONNECTION_ID.eq(CONNECTION.ID))
         .where(CONNECTION_OPERATION.OPERATION_ID.eq(operationId))).fetch();
-    return getStandardSyncsFromResult(result);
+
+    final List<UUID> connectionIds = connectionResult.stream().map(r -> r.get(CONNECTION.ID)).toList();
+
+    final Result<Record> connectionOperationResult = database.query(ctx -> ctx.select(CONNECTION_OPERATION.asterisk()))
+        .from(CONNECTION_OPERATION)
+        .where(CONNECTION_OPERATION.CONNECTION_ID.in(connectionIds))
+        .fetch();
+
+    return getStandardSyncsFromResult(connectionResult, connectionOperationResult);
   }
 
   public List<StandardSync> listWorkspaceStandardSyncs(final UUID workspaceId, final boolean includeDeleted) throws IOException {
-    final Result<Record> result = database.query(ctx -> ctx.select(CONNECTION.asterisk())
+    final Result<Record> connectionResult = database.query(ctx -> ctx.select(CONNECTION.asterisk())
         .from(CONNECTION)
         .join(ACTOR).on(CONNECTION.SOURCE_ID.eq(ACTOR.ID))
         .where(ACTOR.WORKSPACE_ID.eq(workspaceId)
             .and(includeDeleted ? noCondition() : CONNECTION.STATUS.notEqual(StatusType.deprecated))))
         .fetch();
-    return getStandardSyncsFromResult(result);
+
+    final List<UUID> connectionIds = connectionResult.stream().map(r -> r.get(CONNECTION.ID)).toList();
+
+    final Result<Record> connectionOperationResult = database.query(ctx -> ctx.select(CONNECTION_OPERATION.asterisk()))
+        .from(CONNECTION_OPERATION)
+        .where(CONNECTION_OPERATION.CONNECTION_ID.in(connectionIds))
+        .fetch();
+
+    return getStandardSyncsFromResult(connectionResult, connectionOperationResult);
   }
 
-  private List<StandardSync> getStandardSyncsFromResult(final Result<Record> result) throws IOException {
+  private List<StandardSync> getStandardSyncsFromResult(
```

Comment:
can we add a javadocs here? The method name makes it seem like it's a POJO transformation, but we are actually do more logic under the hood.